.github/workflows/scan.yml

```diff
@@ -0,0 +1,48 @@
+name: scan-images
+
+on:
+  schedule:
+    - cron: "0 12 * * 1"
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  scan:
+    name: Trivy
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+    - name: Make images
+      run: make REGISTRY=gcr.io/k8s-staging-cluster-api PULL_POLICY=IfNotPresent TAG=dev ARCH=amd64 docker-build
+    - name: Run Trivy vulnerability scanner on clusterctl image
```

Comment:
We can start with that but essentially the job will fail on the first image with vulnerabilities and it will also fail even if there are now new versions of dependencies which have the issues fixed.

I kind of liked how they did it here: https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/Makefile#L155-L161 (shown all vulnerabilities of everything and then only failing for the ones that can be fixed) as I suggested here: https://github.com/kubernetes-sigs/cluster-api/pull/7604#discussion_r1031038192

As soon as there's an unfixable vulnerability we will have to fallback to running the script manually (and modifying it so we can scan more images than just until the first one fails)